src/libraries/Microsoft.PowerFx.Interpreter/UDFHelper.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Interpreter
+{
+    internal class UDFHelper
+    {
+        public class LazyCheck
+        {
+            internal string _expressionText;
+            internal RecordType _parameterType;
+            internal ParserOptions _options;
+            internal Engine _engine;
+
+            public LazyCheck(Engine engine, string expressionText, RecordType parameterType = null, ParserOptions options = null)
```

Comment:
>RecordType parameterType = null, ParserOptions options = null [](http://example.com/codeflow?start=67&length=61)

Are these/should these be optional?

Reply:
I honestly don't know, I based these off of Engine#Check, it isn't necessary for the specific use of LazyCheck that I am using right now, where it just takes in a string.

Reply:
Removed ParserOptions as a parameter but left RecordType as it's used.